entitycollection.go

```diff
@@ -35,6 +35,45 @@ func (ec *EntityCollection) AddEntity(entity *Entity) error {
 	return nil
 }
 
+func (ec *EntityCollection) AddEntityFromMap(data map[string]any) error {
+	entity := NewEntity()
+
+	// get metadata
+	if id, found := data["id"]; found {
```

Comment:
What if key is capital ID?

Reply:
Good question. The assumption or use case was when you have serialised JSON from an entity and have loaded it into a map rather than using a parser. I think this can be addressed by the docs, if you dont use these keys then the entity wont be populated in the way you expect.